### ISSUE TITLE
dropbox_oauth2 redirect-url has hard-coded https

### ISSUE THREAD

**Issue by saqib-cliqz** [pennersr/django-allauth#1646]
Hi,

As described in the docs, I have set the Development callback URL to 
```http://localhost:8000/accounts/dropbox_oauth2/login/callback/```

However, I am getting this error: 
```
Invalid redirect_uri: "https://localhost:8000/accounts/dropbox_oauth2/login/callback/": It must exactly match one of the redirect URIs you've pre-configured for your app (including the path).
```

After looking at the code, it turns out that in 
```django-allauth/allauth/socialaccount/providers/dropbox_oauth2/views.py``` , redirect_url_protocol is hard-coded to https

```python
class DropboxOAuth2Adapter(OAuth2Adapter):
    provider_id = DropboxOAuth2Provider.id
    access_token_url = 'https://api.dropbox.com/1/oauth2/token'
    authorize_url = 'https://www.dropbox.com/1/oauth2/authorize'
    profile_url = 'https://api.dropbox.com/1/account/info'
    redirect_uri_protocol = 'https'
```

My question is, how can I overwrite it to use http for local testing purposes?

**Comment by lesha1201** [comment]
@pennersr Sorry for pinging but any updates on this? Dropbox seems to support HTTP for localhost. Do you know why `https` were hardcoded for Dropbox?

**Comment by pennersr** [comment]
Closed via 4a404c17

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/1646
